sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -222,6 +222,20 @@ abstract class Expression extends TreeNode[Expression] {
    */
   def childrenResolved: Boolean = children.forall(_.resolved)
 
+  // Return a roughly canonicalized expression, which recursively canonicalizes each expression node
+  // without changing the structure of the expression tree.
```

Comment:
Can we also describe when this can be used? If somebody mistakenly uses `roughlyCanonicalized` instead of `canonicalized`, does it still work?

Reply:
I meant that

> This method should only be called in `canonicalized` or subclass's `roughlyCanonicalized`.

doesn't explain when and why do we use this.